### PR TITLE
refactor: use `Validation` in Bootstrap

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/5671

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 1]
```diff
@@ -209,7 +209,7 @@ object Main {
           Bootstrap.bootstrap(cwd)(System.out) match {
             case Result.Ok(bootstrap) =>
               val result = bootstrap.test(o)
-              System.exit(getCode(result))
+              System.exit(toCode(result))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:212]: But if we are always under an OK branch, we might as well, just call exit(0)

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/Main.scala:212]: I think the Bootstrap is Ok but then the method called on the Bootstrap object can also fail (within the Ok branch)

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 2]
```diff
@@ -139,85 +143,97 @@ object Main {
           System.exit(getCode(result))
 
         case Command.Init =>
-          val result = Bootstrap.init(cwd, options)(System.out)
-          System.exit(getCode(result))
+          Bootstrap.init(cwd, options)(System.out) match {
+            case Result.Ok(_) =>
+              System.exit(EXIT_SUCCESS)
+            case Result.Err(e) =>
+              System.err.println(e)
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/Main.scala:150]: Is it correct to print these errors? (See `printErrors` below)

**Comment by magnus-madsen** [comment]
I think we need to:

- Switch to `Validation` as the return type.
- Introduce a new error type so we are not returning strings.

@jaschdoc Before any programming, does this seem feasible?

**Comment by jaschdoc** [comment]
Yeah that seems doable

**Comment by magnus-madsen** [comment]
What's the status on this?

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 3]
```diff
@@ -32,6 +32,10 @@ import java.nio.file.Paths
   */
 object Main {
 
+  private val EXIT_SUCCESS = 0
+  private val EXIT_FAILURE = 1
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:36]: I dont like these. Let us just inline them. Everyone should know that 0 is success :)

**Comment by jaschdoc** [comment]
> What's the status on this?

I have to refactor this to Validation but I'll do that soon

**Comment by jaschdoc** [comment]
Now uses Validation and has error types that are not just strings

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 4]
```diff
@@ -258,9 +272,9 @@ object Main {
   /**
     * Extracts the exit code from the given result.
     */
-  private def getCode(result: Result[_, Int]): Int = result match {
+  private def getCode[T, E](result: Result[T, E]): Int = result match {
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/Main.scala:275]: Maybe this function should be removed. I'll have to double check if it's still used somewhere

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/Main.scala:275]: Yes, lets remove it.

**Pull request merged**